### DISCUSSION TITLE
How to show `tabset` in full window with?

### DISCUSSION THREAD

**Discussion by hopezh** [quarto-dev/quarto-cli#8335] [Q&A] [answered]
### Description

In the example `qmd` file below, the first `python` code block and the output of the other two code blocks in a `tabset` can be rendered in full window width, using the `#| column: screen` option. 

However, the `tabset` itself still occupies only the middle part of a page.

May I ask if there is a way to make the `tabset` to expend to full screen width, too? 

Thanks.

Image of the rendered page:
![2024-01-18-14-26-03](https://github.com/quarto-dev/quarto-cli/assets/23435810/f0cb9047-1017-4a77-9c39-b9f168eeb7f3)

The `test.qmd` file:

```
---
format: 
    html:
        page-layout: full
---

# Calculate num of modules along two edges of each module cluster outline

\```{python}
#| column: screen
import math
import numpy as np
import pandas as pd
import polars as pl
print('numpy:', np.__version__)
print('pandas:', pd.__version__)
print('polars:', pl.__version__)
\```

Read the data.

::: {.panel-tabset}

## polars

\```{python}
#| column: screen
csv_path = './1 tw_taizhong_sanying/test_L2P.csv'

df = pl.read_csv(csv_path)
df
\```

## pandas


\```{python}
#| column: screen
csv_path = './1 tw_taizhong_sanying/test_L2P.csv'

df = pd.read_csv(csv_path)
df.head()
\```

:::
```

**Comment by mcanouil** [comment]
Could you edit your code to not use `\`? Simply use more backticks to enclose the code, i.e., four in your case.

**Comment by Pierre9344** [comment]
You can do that by having your tabset inside [a column-screen div](https://quarto.org/docs/authoring/article-layout.html#screen-column).

````
::: {.column-screen}
::: {.panel-tabset}

## polars

```{python}
#| column: screen
csv_path = './1 tw_taizhong_sanying/test_L2P.csv'

df = pl.read_csv(csv_path)
df
```

## pandas

```{python}
#| column: screen
csv_path = './1 tw_taizhong_sanying/test_L2P.csv'

df = pd.read_csv(csv_path)
df.head()
```

:::

:::
````

**Reply by mcanouil** [reply, marked as answer]
A bit of explanation here:

`#| column: screen` controls the code cells and outputs display which is conditional to its surroundings.
If you want the tabset to span full screen, then as showed you need to tell Quarto that the tabset should span full screen.

By default code cells will occupy the available space, so `#| column: screen` becomes useless.
The following is enough:

````qmd
---
title: "Quarto Playground"
format: html
---

::: {.column-screen}
::: {.panel-tabset}

## polars

```{python}
csv_path = './1 tw_taizhong_sanying/test_L2P.csv'

df = pl.read_csv(csv_path)
df
```

## pandas

```{python}
csv_path = './1 tw_taizhong_sanying/test_L2P.csv'

df = pd.read_csv(csv_path)
df.head()
```

:::

:::
````

**Reply by hopezh** [reply]
Thank you @Pierre9344 and @mcanouil  very much!
